### PR TITLE
Logging updates

### PR DESCRIPTION
**Pull request by maradionov**
Add new log level METRIC (between INFO and WARNING) to print only metric information. Send metrics to tensorboard. Add example to PyTorch MNIST notebook.

### CODE REVIEW

File: openfl-workspace/default/plan/plan.yaml [hunk 1]
```diff
@@ -9,6 +9,7 @@ aggregator :
     best_state_path : save/best.pbuf
     last_state_path : save/last.pbuf
     rounds_to_train : 10
+    tensor_board    : false
```
Comment (itrushkin) [inline openfl-workspace/default/plan/plan.yaml]:
```suggestion
    tensorboard    : false
```
We probably should preserve naming from [tensorflow/tensorboard](https://github.com/tensorflow/tensorboard) and [lanpa/tensorboardX](https://github.com/lanpa/tensorboardx).

File: openfl-workspace/torch_cnn_mnist/plan/plan.yaml [hunk 2]
```diff
@@ -9,6 +9,7 @@ aggregator :
     best_state_path : save/torch_cnn_mnist_best.pbuf
     last_state_path : save/torch_cnn_mnist_last.pbuf
     rounds_to_train : 10
+    tensor_board    : true
```
Comment (itrushkin) [inline openfl-workspace/torch_cnn_mnist/plan/plan.yaml]:
```suggestion
    tensorboard    : true
```

File: openfl/component/aggregator/aggregator.py [hunk 3]
```diff
@@ -103,6 +105,7 @@ def __init__(self,
         self.collaborator_tasks_results = {}
         # {TaskResultKey: data_size}
         self.collaborator_task_weight = {}
+        self.tb_writer = SummaryWriter(self.log_dir, flush_secs=10) if tensor_board else None
```
Comment (itrushkin) [inline openfl/component/aggregator/aggregator.py]: Could users pass `SummaryWriter` instance instead of the flag deciding the TensorBoard usage? It would allow users to define the logging directory, flush frequency, and other keyword arguments of `SummaryWriter`. Moreover, in this case, we do not need `tensorboardX` as a dependency.

File: openfl/component/aggregator/aggregator.py [hunk 4]
```diff
@@ -41,6 +42,7 @@ def __init__(self,
                  assigner,
 
                  rounds_to_train=256,
+                 tensor_board=False,
```
Comment (maradionov) [inline openfl/component/aggregator/aggregator.py]:
```suggestion
                 tensorboard=False,
```

File: openfl/component/aggregator/aggregator.py (hunk 3 above)
Comment (maradionov) [inline openfl/component/aggregator/aggregator.py]:
```suggestion
        self.tb_writer = SummaryWriter(self.log_dir, flush_secs=10) if tensorboard else None
```

File: openfl/utilities/utils.py [hunk 5]
```diff
@@ -3,6 +3,40 @@
...
+    Args:
+        levelName: name of log level.
+        levelNum: log level value.
+        methodName: log method wich will use new log level (default = levelName.lower())
+
+    """
+    if not methodName:
+        methodName = levelName.lower()
+
+    if hasattr(logging, levelName):
```
Comment (aleksandr-mokrov) [inline openfl/utilities/utils.py]: It looks redundant.  We don't catch this exception and I think that it is not a problem if we would try add the same level twiсe into logging. In this case, it is enough to exit.

File: openfl-workspace/default/plan/plan.yaml (hunk 1 above)
Comment (alexey-gruzdev) [reply openfl-workspace/default/plan/plan.yaml]: Agree with @itrushkin for naming

File: openfl/utilities/utils.py (hunk 5 above)
Comment (maradionov) [reply openfl/utilities/utils.py]: done 3ed9f513adc09160ee04a63eb3e910b3b0fdfa65

File: openfl-workspace/default/plan/plan.yaml [hunk 6]
```diff
@@ -9,6 +9,7 @@ aggregator :
     best_state_path : save/best.pbuf
     last_state_path : save/last.pbuf
     rounds_to_train : 10
+    tensorboard    : false
```
Comment (dmitryagapov) [inline openfl-workspace/default/plan/plan.yaml]: align familiare with other strings

File: openfl-workspace/torch_cnn_mnist/plan/plan.yaml [hunk 7]
```diff
@@ -9,6 +9,7 @@ aggregator :
     best_state_path : save/torch_cnn_mnist_best.pbuf
     last_state_path : save/torch_cnn_mnist_last.pbuf
     rounds_to_train : 10
+    tensorboard    : true
```
Comment (dmitryagapov) [inline openfl-workspace/torch_cnn_mnist/plan/plan.yaml]: align familiar with other strings

File: openfl/interface/cli.py [hunk 8]
```diff
@@ -7,35 +7,40 @@
 from click import echo, option, pass_context, style
 from sys import argv
 from pathlib import Path
+import os
 
+from openfl.utilities import addLoggingLevel
```
Comment (dmitryagapov) [inline openfl/interface/cli.py]: https://www.python.org/dev/peps/pep-0008/#imports
Imports should be grouped in the following order:

Standard library imports.
Related third party imports.
Local application/library specific imports.
You should put a blank line between each group of imports.

The same recommendation for all imports.

File: openfl/interface/cli.py [hunk 9]
```diff
@@ -7,35 +7,40 @@
...
-            'error': ERROR,
-            'critical': CRITICAL
-        }
+    if isinstance(level, str):
+        level = level.upper()
 
-    if level.lower() in ['debug', 'error']:
-        os.environ['GRPC_VERBOSITY'] = level.upper()
+    if level in ['DEBUG', 'ERROR']:
+        os.environ['GRPC_VERBOSITY'] = level
```
Comment (dmitryagapov) [inline openfl/interface/cli.py]: When you use isinstance it looks like you want to handle both types of level `int` and `str`, as it used in logging module, but here only `str` expected and handeled on line 29.

File: openfl/native/native.py [hunk 10]
```diff
@@ -39,12 +41,13 @@ def setup_plan(log_level='CRITICAL'):
     cols_config = 'plan/cols.yaml'
     data_config = 'plan/data.yaml'
 
+    curent_level = logging.root.level
```
Comment (dmitryagapov) [inline openfl/native/native.py]: *current :)

File: openfl/utilities/utils.py [hunk 11]
```diff
@@ -3,6 +3,33 @@
 """Utilities module."""
 
 import numpy as np
+import logging
+
+
+def addLoggingLevel(levelName, levelNum, methodName=None):
```
Comment (dmitryagapov) [inline openfl/utilities/utils.py]: https://www.python.org/dev/peps/pep-0008/#function-and-variable-names

**Comment by alexey-gruzdev** [comment]
@maradionov let's also check how it would work with new interactive API.

File: openfl/component/aggregator/aggregator.py (hunk 3 above)
Comment (maradionov) [reply openfl/component/aggregator/aggregator.py]: resolve 747b5c53fee5da2143a617db0311e523c7c937d9

File: openfl/interface/cli.py (hunk 9 above)
Comment (maradionov) [reply openfl/interface/cli.py]: solved 95fcf3916295687c77d7d34fc04074389b034e59

File: openfl/interface/cli.py (hunk 8 above)
Comment (maradionov) [reply openfl/interface/cli.py]: solved 79e053176690b2e97bcb12e9e5a66314d211dc02

File: openfl/utilities/utils.py (hunk 11 above)
Comment (maradionov) [reply openfl/utilities/utils.py]: solved 79e053176690b2e97bcb12e9e5a66314d211dc02

File: openfl/native/native.py (hunk 10 above)
Comment (maradionov) [reply openfl/native/native.py]: solved 79e053176690b2e97bcb12e9e5a66314d211dc02

**Comment by alexey-gruzdev** [comment]
Closed this since we've merged it to #104
